x/wasm/ioutils/ioutil.go

```diff
@@ -11,8 +11,8 @@ import (
 )
 
 // Uncompress expects a valid gzip source to unpack or fails. See IsGzip
-func Uncompress(gzipSrc []byte, limit uint64) ([]byte, error) {
-	if uint64(len(gzipSrc)) > limit {
+func Uncompress(gzipSrc []byte, limit int64) ([]byte, error) {
+	if int64(len(gzipSrc)) > limit {
```

Comment:
Why did this become a signed int? Just wondering, as it was approved in the other PR already.

Reply:
Because ReadAll method requires an int64 and also MaxWasmSize  is an integer. So it didn't make sense to do conversions int -> uint before calling the Uncompress method and uint -> int inside the method before calling ReadAll. This also avoids all possible risks related to uint -> int conversion.